File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int dup = x;
        long long revNum = 0;
        if(x < 0)   return false;
        while(x != 0){
            int rem = x%10;
            revNum = (revNum*10) + rem;
            x = x/10;
        }
        if(dup == revNum)   return true;
        return false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String numStr = Integer.toString(x);
        int i = 0, j = numStr.length()-1;
        while(i < j){
            if(numStr.charAt(i) != numStr.charAt(j))    return false;
            i++; j--;
        }
        return true;
    }
}